src/main/java/sirius/biz/jobs/batch/file/FileImportJob.java

```diff
@@ -42,11 +51,68 @@ protected FileImportJob(VirtualObjectParameter fileParameter, ProcessContext pro
     @Override
     public void execute() throws Exception {
         VirtualObject file = process.require(fileParameter);
+
         try (InputStream in = storage.getData(file)) {
+            if (!canHandleFileExtension(file.getFileExtension()) && "zip".equalsIgnoreCase(file.getFileExtension())) {
+                executeForZippedFile(in);
+
+                return;
+            }
+
+            if (!canHandleFileExtension(file.getFileExtension())) {
+                throw Exceptions.createHandled().withNLSKey("FileImportJob.fileNotSupported").handle();
+            }
+
             executeForStream(file.getFilename(), in);
         }
     }
 
+    /**
+     * Performs the import for a zipped file.
+     *
+     * @param in the data to import
+     * @throws Exception in case of an error during the import
+     */
+    protected void executeForZippedFile(InputStream in) throws Exception {
+        process.log(ProcessLog.info().withNLSKey("FileImportJob.importingZipFile"));
+
+        try (ZipInputStream zipInputStream = new ZipInputStream(in)) {
+            boolean fileFound = false;
+            ZipEntry entry = zipInputStream.getNextEntry();
+
+            while (entry != null) {
+                if (!isHiddenFile(entry.getName()) && canHandleFileExtension(Files.getFileExtension(entry.getName()))) {
```

Comment:
is this hidden check necessary - wouldn't the file extensions filter be enough?

Reply:
MacOS adds hidden files with the same name and extension to the zip